lib/facter/facts/solaris/virtual.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+module Facts
+  module Solaris
+    class Virtual
+      FACT_NAME = 'virtual'
+
+      def initialize
+        @log = Facter::Log.new(self)
+      end
+
+      def call_the_resolver
+        @log.debug('Solaris Virtual Resolver')
+
+        fact_value = check_ldom || check_zone || 'physical'
```

Comment:
if ldom and zone fail to retrieve the fact this method is called => https://github.com/puppetlabs/facter/blob/fd44acffcc64544da913e49acfa02bf0da7be62c/lib/src/facts/solaris/virtualization_resolver.cc#L36  before stating that is physical. 

Methos definition : https://github.com/puppetlabs/facter/blob/2aa2d1cd6487f73fb8e311cf027dcf30c2166e28/lib/src/facts/resolvers/virtualization_resolver.cc#L74
this is already implementing in linux/virtual (check_other_facts method)

Reply:
Updated, thanks!